### PR TITLE
feat: Max Samples for thumbnails for moderation

### PR DESCRIPTION
**Pull request by philcluff**
Adds a maxSamples option to control the number of thumbnails sampled during moderation workflows. This allows users to balance moderation thoroughness against API costs.

This PR supersedes #90 

Changes

  - src/primitives/thumbnails.ts: Added maxSamples parameter to ThumbnailOptions interface
    - When set, caps the number of generated thumbnails to the specified limit
    - Samples are evenly distributed across the video duration
    - First frame (t=0) and last frame (t=duration) are always included when maxSamples >= 2
  - src/workflows/moderation.ts: Added maxSamples parameter to ModerationOptions interface
    - Passes through to getThumbnailUrls for thumbnail-based moderation
    - Does not affect transcript-based moderation (audio-only assets)

Usage

```
  // Default: samples every 10 seconds (100s video = 10 thumbnails)
  await getModerationScores(assetId, { thumbnailInterval: 10 });

  // With maxSamples: caps at 5 evenly-distributed samples
  await getModerationScores(assetId, {
    thumbnailInterval: 10,
    maxSamples: 5  // Samples at t=0, 25, 50, 75, 100
  });
```

Testing

  - Unit tests (tests/unit/thumbnails.test.ts): 17 tests covering thumbnail generation logic, maxSamples behavior, edge cases, and timestamp distribution
  - Integration tests (tests/integration/moderation.test.ts): 6 tests verifying moderation works correctly with limited samples across OpenAI and Hive providers
  - Signed playback tests (tests/integration/signed-playback.test.ts): 1 test confirming maxSamples works with signed URLs

### CODE REVIEW

File: tests/integration/moderation.test.ts [hunk 1]
```diff
@@ -154,4 +154,100 @@ describe("moderation Integration Tests", () => {
...
+    it("should work with very small maxSamples", async () => {
+      const result = await getModerationScores(safeAsset, {
+        provider: "openai",
+        model: "omni-moderation-latest",
+        maxSamples: 2,
+      });
+
+      expect(result).toBeDefined();
+      expect(result.thumbnailScores.length).toBe(2);
+      expect(result.thumbnailScores.filter(s => !s.error).length).toBeGreaterThan(0);
```
[the remark this answers is not in the document]
Comment (philcluff) [reply tests/integration/moderation.test.ts:202]: Current version is consistent with the rest of the code bacse.

File: tests/unit/thumbnails.test.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+      // Middle timestamps should be evenly spaced
+      // Expected spacing: 200 / (10-1) = ~22.22
+      for (let i = 1; i < timestamps.length - 1; i++) {
+        const expected = (200 / (10 - 1)) * i;
+        expect(timestamps[i]).toBeCloseTo(expected, 1);
+      }
+    });
+  });
+
+  describe("uRL formatting", () => {
```
[the remark this answers is not in the document]
Comment (philcluff) [reply tests/unit/thumbnails.test.ts]: Nope, or ESLint fails.

**Review by monsieurBoutte — APPROVED** [review]
